### DISCUSSION TITLE
Request support for access serialize data with dot notation

### DISCUSSION THREAD

**Discussion by tumzied** [encode/django-rest-framework#8874] [General]
currently, `Serializer.data` property return python dictionary, it would be much cleaner code if  `Serializer.data` return `dataclass` instance of that class instate of dictionary, So that we could access serialize data with dot notation, the source-code editor could show better suggestion and write more clean code.

```python

from rest_framework import serializers

class UserSerializer(serializers.Serializer):
    name = serializers.CharField(max_length=500)
    phone = serializers.CharField(max_length=10)


def my_view(request):
   serializer = UserSerializer(data=request.data)
   serializer.is_valid()
   data = serializer.data
   print(data['name']) # current implementation 
   # print(data.name) # access with dot notation  ,  i want this
```

**Comment by auvipy** [comment]
could be a possibility if achieving this don't change much from architectural point of view
